### PR TITLE
Add a start time argument to predict calls

### PR DESCRIPTION
**Pull request by ajbozarth**
I have added an optional `start_time` argument to the predict call. This will have the model run prediction at the given start time rather than the start of the audio file.

This was inspired by the latest updates to my WIP web app based on this model https://github.com/ajbozarth/MAX-Audio-Classifier-Web-App

Implementation notes:

- The given time stamp (`start_time`) is converted to an relative embedding index by dividing by `vggish_params.EXAMPLE_HOP_SECONDS` (=`0.96`) and setting to `int`. This is to match an adjustment in the embedding process where one embedding = 0.96 seconds.
- If `start_time=0` then the if/elif statement is skipped and the original code is run as is
- If `start_time` is either negative or larger than the audio file length the a `ValueError` is thrown and the converted to a `BadRequest` to return to the user
- If `start_time` is within the final 10s of the file then it will loop the given section the same as if it was a <10s file
- `start_time` is a float in order to more easily support the value of HTML5 audio `currentTime` that I plan to use in my web app.
- Due to floating point arithmetic, negative values greater than `-0.96` will be accepted as valid since they'll round to 0 in the code, since the edge case detection is run after the time stamp is set to int it does not affect edge cases in the actual code and is only an oddity from a user's perspective

### CODE REVIEW

**Comment by ajbozarth** [comment]
Again noting this is just a WIP proposal that I'm open to feedback on.

File: core/backend.py [hunk 1]
```diff
@@ -96,6 +96,9 @@ def classifier_pre_process(self, embeddings):
         Returns:
             embeddings = numpy array of shape (1,10,128), dtype=float32.
         """
+        if 0 < time_stamp < embeddings.shape[0]:
+            embeddings = embeddings[time_stamp:time_stamp+10,:]
```
Comment (bdwyer2) [inline core/backend.py]: Maybe you should add a warning here if the `if` statement is false

Comment (ajbozarth) [reply core/backend.py]: Yeah, I just didn't know how to surface it properly atm, but that was my best idea too

**Comment by kmh4321** [comment]
@ajbozarth I have noticed that it rounds up to the nearest second. So basically if your audio is 12.3 seconds, `embeddings.shape[0]` would be 13. I have however not tested it with 3+ minutes and not sure why 8 seconds is being added to the clip. It might be better to clip the relevant audio portion outside the wrapper using FFmpeg. It will also be much faster because you won't be generating embeddings for unnecessary portions.

**Comment by ajbozarth** [comment]
Thanks for the feedback @kmh4321 I'll go back to my ffmpeg implementation then

**Comment by ajbozarth** [comment]
Based on a conversation I had with @frreiss we managed to track down why the `embeddings.shape` value was off, specifically each embedding represents 0.96 seconds not 1. I have no thoroughly tested my new code, but it at least fixes that one issue.

Open Questions needing feedback:
- How are my comments and variable names
- Should the clip/crop functionality be moved into the embedding code to minimize the embeddings created?

**Comment by ajbozarth** [comment]
I've also updated the implementation notes in the description to match the current implementation

**Comment by ajbozarth** [comment]
@MLnick @bdwyer2 now that we're not busy on the latest models, can you give this another look? If/once it's merged I'll be able to start reworking my audio web app (when I have time)

File: api/model.py [hunk 2]
```diff
@@ -39,6 +39,7 @@ def get(self):
 audio_parser = api.parser()
 audio_parser.add_argument('audio', type=FileStorage, location='files', required=True,
                           help="signed 16-bit PCM WAV audio file")
+audio_parser.add_argument('start_time', type=float, default=0, help='The time (in seconds) in the audio file to run prediction at.')
```
Comment (bdwyer2) [inline api/model.py]: "The time (in seconds)....." --> "The _start_ time (in seconds)"

Comment (bdwyer2) [inline api/model.py]: Maybe the help text could be something like "The number of seconds into the audio file the prediction should start at"

Comment (ajbozarth) [reply api/model.py]: That's fine by me, I'll update it

**Comment by MLnick** [comment]
@ajbozarth changes look good. 

As @kmh4321 mentions it would probably be ideal to do clipping of larger audio files before generating embeddings, as on a clip of a few minutes I think it will make a very large difference in performance.

Not sure how easy that will be though - we could go ahead with this and make that change later if required.

**Comment by ajbozarth** [comment]
@MLnick I did look into that a bit and it would not be as easy to implement as this since rather than slicing an array of embeddings we would have to actually edit the audio file itself, which would require a audio editing library. Given the response time I get on a standard song file when testing this PR I'm not sure the speed up is needed. (You can open up an issue to address it later if you think it's still worth it)

**Pull request merged**